{Storage} Doc for storage data plane auth (#22104)
Azure/azure-cli: doc/storage_data_plane_commands_auth.md

# Storage Data Plane Commands Auth

> Azure operations can be divided into two categories - control plane and data plane. By running storage data plane commands, requests will be sent directly to Storage service instead of being sent to ARM(Azure Resource Manager). Hence additional credentials are required for successfully auth to Storage service.

Reviewer: We usually call it mgmt-plane and data-plane, don't we?

Author: I want to follow the doc I shared in `See More` section: https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/control-plane-and-data-plane

> Azure operations can be divided into two categories - control plane and data plane. This article describes the differences between those two types of operations.
> 
> You use the control plane to manage resources in your subscription. You use the data plane to use capabilities exposed by your instance of a resource type.

> Azure operations can be divided into two categories - control plane and data plane. By running storage data plane commands, requests will be sent directly to Storage service instead of being sent to ARM(Azure Resource Manager). Hence additional credentials are required for successfully auth to Storage service.

Reviewer: ```suggestion
Azure operations can be divided into two categories - control plane and data plane. By running storage data plane commands, requests will be sent directly to Storage service instead of being sent to ARM (Azure Resource Manager). Hence additional credentials are required for successful authentication to Storage service.
```

## Authorization Options

> There are several ways to authorize for storage data plane commands:

Reviewer: ```suggestion
There are several ways to authenticate to storage data plane commands:
```

See https://english.stackexchange.com/questions/225756/is-it-authenticate-with-or-authenticate-against

### Azure Active Directory integration

> By specifying `--auth-mode login`, CLI will use the OAuth token returned by AAD(Azure Active Directory) to authorize requests against Storage service.

Reviewer: ```suggestion
By specifying `--auth-mode login`, CLI will use the OAuth token returned by AAD (Azure Active Directory) to authenticate to Storage service.
```

Here's an example of getting properties of a blob in a storage account with AAD token
```cmd
# Get blob properties using AAD token

> > az storage blob show --name $blobName --container-name $containerName --account-name $accountName --auth-mode login

Reviewer: ```suggestion
az storage blob show --name $blobName --container-name $containerName --account-name $accountName --auth-mode login
```

The prompt `>` is only for Windows. Bash uses `$`.

```

> ### Shared key authorization

Reviewer: ```suggestion
### Shared key authentication
```

By specifying `--account-key $accountKey` or `--connection-string $connectionString` configured using account key, CLI can use Shared Key by signing each request with the storage account access key.

> Here's an example of getting properties of a blob in a storage account with account key

Reviewer: Better to have a comma here. Same below.

```suggestion
Here's an example of getting properties of a blob in a storage account with account key:
```

```cmd
# Get storage account access key

> > $accountKey = az storage account keys list --account-name $accountName --resource-group $rgName --query "[0].value"

Reviewer: By default, we provide multiple-command script in Bash (which is the default of Cloud Shell). If this is for PowerShell, we'd better explicitly call that out.

Author: Changed to Bash now

# Get blob properties with account key
> az storage blob show --name $blobName --container-name $containerName --account-name $accountName --account-key $accountKey
```

> Here's an example of getting properties of a blob in a storage account with connection string configured using account key

Reviewer: ```suggestion
Here's an example of getting properties of a blob in a storage account with connection string configured using account key:
```

```cmd
# Get storage account access key
> $accountKey = az storage account keys list --account-name $accountName --resource-group $rgName --query "[0].value"
# Configure a connection string
> $connectionString = "DefaultEndpointsProtocol=https;AccountName=$accountName;AccountKey=$accountKey"
# Get blob properties with connection string
> az storage blob show --name $blobName --container-name $containerName --connection-string $connectionString
```

### Shared access signatures

> By specifying `--sas-token $sasToken` or `--connection-string $connectionString` configured using shared access signature(SAS), CLI can get a signed URI including a token that indicates how the resources may be accessed by the client.

Reviewer: ```suggestion
By specifying `--sas-token $sasToken` or `--connection-string $connectionString` configured using shared access signature (SAS), CLI can get a signed URI including a token that indicates how the resources may be accessed by the client.
```

Edge's grammar check warns us:

![image](https://user-images.githubusercontent.com/4003950/164161119-225e179f-039e-41ec-a5aa-f2322ebd1c18.png)
